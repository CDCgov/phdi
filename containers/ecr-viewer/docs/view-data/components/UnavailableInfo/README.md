[**ecr-viewer**](../../../README.md) â€¢ **Docs**

***

[ecr-viewer](../../../README.md) / view-data/components/UnavailableInfo

# view-data/components/UnavailableInfo

## Index

### Functions

- [default](functions/default.md)
