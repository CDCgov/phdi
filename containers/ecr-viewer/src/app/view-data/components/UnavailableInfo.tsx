import { DisplayData } from "@/app/utils";
import {
  AccordianSection,
  AccordianH3,
  AccordianDiv,
} from "../component-utils";

interface UnavailableInfoProps {
  demographicsUnavailableData: DisplayData[];
  socialUnavailableData: DisplayData[];
  encounterUnavailableData: DisplayData[];
  clinicalUnavailableData: DisplayData[];
  providerUnavailableData: DisplayData[];
  activeProblemsUnavailableData: DisplayData[];
}

const UnavailableInfo = ({
  demographicsUnavailableData,
  socialUnavailableData,
  encounterUnavailableData,
  clinicalUnavailableData,
  providerUnavailableData,
  activeProblemsUnavailableData,
}: UnavailableInfoProps) => {
  const renderRow = (item: any, index: number) => {
    return (
      <div key={index}>
        <div className="grid-row">
          <div className="data-title font-sans-xs">
            <h4>{item.title}</h4>
          </div>
          <div className="grid-col-auto font-sans-xs">N/A</div>
        </div>
        <div className={"section__line_gray"} />
      </div>
    );
  };
  const renderSection = (sectionTitle: string, data: DisplayData[]) => {
    return (
      <div className="margin-bottom-4">
        <h3
          className="usa-summary-box__heading padding-bottom-205"
          id="summary-box-key-information"
        >
          {sectionTitle}
        </h3>
        <div className="usa-summary-box__text">
          {data.map((item, index) => renderRow(item, index))}
        </div>
      </div>
    );
  };

  return (
    <AccordianSection>
      {demographicsUnavailableData?.length > 0 &&
        renderSection("Demographics", demographicsUnavailableData)}
      {socialUnavailableData?.length > 0 &&
        renderSection("Social History", socialUnavailableData)}
      {encounterUnavailableData?.length > 0 &&
        renderSection("Encounter Details", encounterUnavailableData)}
      {clinicalUnavailableData.length > 0 &&
        renderSection("Clinical Info", clinicalUnavailableData)}
      {providerUnavailableData.length > 0 &&
        renderSection("Provider Details", providerUnavailableData)}
      {activeProblemsUnavailableData?.length > 0 && // Add other items under symptoms and problems here
        renderSection("Symptoms and Problems", activeProblemsUnavailableData)}
    </AccordianSection>
  );
};

export default UnavailableInfo;
