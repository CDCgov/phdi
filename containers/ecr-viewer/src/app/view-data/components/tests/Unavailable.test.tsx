import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import UnavailableInfo from "../UnavailableInfo";

describe("UnavailableInfo", () => {
  let container: HTMLElement;
  beforeAll(() => {
    const demographicsUnavailability = [
      {
        title: "Tribal Affiliation",
        value: "",
      },
      {
        title: "Preffered Language",
        value: "",
      },
    ];
    const socialUnavailability = [
      {
        title: "Travel History",
        value: "",
      },
      {
        title: "Pregnancy Status",
        value: "",
      },
      {
        title: "Alcohol Use",
        value: "",
      },
      {
        title: "Sexual Orientation",
        value: "",
      },
      {
        title: "Gender Identity",
        value: "",
      },
    ];
    const encounterUnavailableData = [
      {
        title: "Facility Address",
        value: "",
      },
    ];
    const providerUnavailableData = [
      {
        title: "Provider Name",
        value: "",
      },
    ];
    const vitalUnavailableData = [
      {
        title: "Vitals",
        value: "",
      },
    ];
    container = render(
      <UnavailableInfo
        demographicsUnavailableData={demographicsUnavailability}
        socialUnavailableData={socialUnavailability}
        encounterUnavailableData={encounterUnavailableData}
        providerUnavailableData={providerUnavailableData}
        activeProblemsUnavailableData={[]}
        vitalUnavailableData={vitalUnavailableData}
      />,
    ).container;
  });
  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });
  it("should pass accessibility test", async () => {
    expect(await axe(container)).toHaveNoViolations();
  });
});
