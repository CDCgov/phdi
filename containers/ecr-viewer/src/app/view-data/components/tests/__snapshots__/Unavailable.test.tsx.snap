// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnavailableInfo should match snapshot 1`] = `
<div>
  <div>
    <div
      aria-labelledby="summary-box-key-information"
      class="padding-bottom-3"
    >
      <div
        class="usa-summary-box__body"
      >
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205"
            id="summary-box-key-information"
          >
            Social History
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title font-sans-xs"
                >
                  <h4>
                    Travel History
                  </h4>
                </div>
                <div
                  class="grid-col-auto font-sans-xs"
                >
                  N/A
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title font-sans-xs"
                >
                  <h4>
                    Pregnancy Status
                  </h4>
                </div>
                <div
                  class="grid-col-auto font-sans-xs"
                >
                  N/A
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title font-sans-xs"
                >
                  <h4>
                    Alcohol Use
                  </h4>
                </div>
                <div
                  class="grid-col-auto font-sans-xs"
                >
                  N/A
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title font-sans-xs"
                >
                  <h4>
                    Sexual Orientation
                  </h4>
                </div>
                <div
                  class="grid-col-auto font-sans-xs"
                >
                  N/A
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title font-sans-xs"
                >
                  <h4>
                    Gender Identity
                  </h4>
                </div>
                <div
                  class="grid-col-auto font-sans-xs"
                >
                  N/A
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205"
            id="summary-box-key-information"
          >
            Encounter Details
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title font-sans-xs"
                >
                  <h4>
                    Facility Address
                  </h4>
                </div>
                <div
                  class="grid-col-auto font-sans-xs"
                >
                  N/A
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205"
            id="summary-box-key-information"
          >
            Provider Details
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title font-sans-xs"
                >
                  <h4>
                    Provider Name
                  </h4>
                </div>
                <div
                  class="grid-col-auto font-sans-xs"
                >
                  N/A
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
